indico/modules/rb/client/js/util.jsx

```diff
@@ -234,6 +237,67 @@ export function serializeRecurrenceInfo({type, number, interval}) {
   }
 }
 
+/**
+ * Renders the array of recurrence weekdays into a nicely formatted string
+ * @param {array} weekdays Array of weekdays
+ * @returns {string} Formatted string of weekdays
+ */
+export function renderRecurrenceWeekdays(weekdays) {
+  const order = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+  const weekdaysMap = {
+    mon: 'Monday',
+    tue: 'Tuesday',
+    wed: 'Wednesday',
+    thu: 'Thursday',
+    fri: 'Friday',
+    sat: 'Saturday',
+    sun: 'Sunday',
+  };
+
+  if (!weekdays || weekdays.length === 0) {
+    return null;
+  }
+
+  // order weekdays by their index in the order array
+  weekdays = [...new Set(weekdays)].sort((a, b) => order.indexOf(a) - order.indexOf(b));
+
+  if (weekdays.length === 1) {
+    return (
+      <Translate>
+        Recurs every <Param name="weekday" value={weekdaysMap[weekdays[0]]} />
+      </Translate>
+    );
+  }
+
+  if (weekdays.length === 2) {
+    return (
+      <Translate>
+        Recurs every{' '}
+        <Param
+          name="weekdays"
+          value={`${weekdaysMap[weekdays[0]]} and ${weekdaysMap[weekdays[1]]}`}
+        />
+      </Translate>
+    );
+  }
+
+  // join weekdays with commas and replace last comma with 'and'
+  // (i.e. ['mon', 'wed', 'tue'] -> 'Monday, Tuesday and Wednesday')
+  const weekdaysString = weekdays
+    .map(weekday => weekdaysMap[weekday])
+    .join(', ')
+    .replace(
+      `, ${weekdaysMap[weekdays[weekdays.length - 1]]}`,
+      ` and ${weekdaysMap[weekdays[weekdays.length - 1]]}`
```

Comment:
no! :D

create an array with all but the last weekdays and then join those with the `,` and then use the same logic as for 2 weekdays to show that and the last one